jazz_codeq/index.js

```diff
@@ -0,0 +1,211 @@
+// =========================================================================
+// Copyright � 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const errorHandlerModule = require("./components/error-handler.js"); //Import the error codes module.
+const responseObj = require("./components/response.js"); //Import the response module.
+const configObj = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js"); //Import the logging module.
+const request = require('request');
+const moment = require('moment');
+const messages = require("./config/messages");
+const validation = require('./validation');
+const utils = require('./utils');
+
+function handler(event, context, cb) {
+
+	//Initializations
+	let serviceContext = event;
+	let errorHandler = errorHandlerModule();
+	let config = configObj(serviceContext);
+	
+	logger.init(serviceContext, context);
+
+	try {
+		logger.info(serviceContext.resourcePath);
+
+		utils.getAPIPath(serviceContext.resourcePath)
+		.then(data => {
+			const pathString = data.pathString;
+		
+			logger.info(pathString);
+			//pathstring == 'codeq'
+			if (serviceContext && serviceContext.method && serviceContext.method === 'GET' && pathString === "codeq") {
+				logger.info("code quality service called.");
+
+				const result = getCodeqInputsUsingQuery(serviceContext, config);
+				
+				if (result.error) {
+					const errorMessage = (typeof result.error === 'object') ? JSON.stringify(errorHandler.throwInputValidationError(result.error)) : result.error;
+					return cb(errorMessage);
+				} else {
+					const metrics = result.metrics;
+					const toDate = result.toDate;
+					const fromDate = result.fromDate;
+					const query = result.query;
+					let authToken;

```

Comment:
Limit the scope of `authToken` to the 3rd promise call. I might even get rid of the variable itself.